### PR TITLE
Flatline armor BoS's Nightstalker creation

### PR DESCRIPTION
**Pull request by Moribun**
long story made as short as possible. what we talked about in voice. its a slow heal over time armor craftable by scribes made by Flatline  canonically. heals as much as the trauma pack over time and has an auto-defib when dead because the heal continues like the trauma pack did. its loud and obvious. takes time for the defib to happen and even when it does...takes even longer to heal you out of crit....its not an instant pop up and run...your captured or the armor is removed easily. useful for mobs tho.

## About The Pull Request
^ this. also I added a prototype version to my fluff box instead of follower armor. one so I can present it as an IC thing as a prototype...and two...because I like it for when im off duty. one sprite for a patch work prototype exclusive to the night stalker. and then another sprite and craftable one by BOS scribes...flashy and pretty BOS armor colors

## Why It's Good For The Game
its probably not but I made it anyway.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:
added a 2 heal over time armor that works the same as the trauma backpack
it auto defibs on death but takes alot of time for the slow heal to get them to revivable. and even longer before they can stand. revives on the boarder of dead. and takes a long time to heal out of soft crit
armor stats are armor = list("melee" = 35, "bullet" = 35, "laser" = 30, "energy" = 20, "bomb" = 25, "bio" = 10, "rad" = 10, "fire" = 20, "acid" = 10, "wound" = 10)  (same as initiate armor)
and can only be created by scribes and higher rank scribes.
tightened up my fluff box to be less book saturated because it was fucking with my OCD so I made it just auto-give the traits when selected
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/medium_armor.dm [hunk 1]
```diff
@@ -320,6 +320,87 @@
...
+	. = ..()
+
+/obj/item/clothing/suit/armored/medium/combat/bos_flatline/process()
+	if(iscarbon(loc))
+		var/mob/living/carbon/M = loc
+		if(M.health < M.maxHealth)
+			M.adjustBruteLoss(-2) 
+			M.adjustFireLoss(-2)
+			M.adjustToxLoss(-2)
+			M.adjustOxyLoss(-2)
```
Comment (RadiantFlash7) [inline code/modules/clothing/suits/medium_armor.dm:348]: This can be removed.

File: code/modules/clothing/suits/medium_armor.dm [hunk 2]
```diff
@@ -320,6 +320,87 @@
...
+/obj/item/clothing/suit/armored/medium/combat/bos_flatline/process()
+	if(iscarbon(loc))
+		var/mob/living/carbon/M = loc
+		if(M.stat == DEAD)
```
Comment (RadiantFlash7) [inline code/modules/clothing/suits/medium_armor.dm:352]: Make this if(M.stat == DEAD || M.health < M.maxhealth)

**Comment by Moribun** [comment]
its done. if the armor needs anymore balancing carl lemme know. dropped the rating for it to initiate armor value like radiant suggested and tightened up the code with rayfords help so its less bloaty

File: modular_citadel/code/modules/client/loadout/__donator.dm [hunk 3]
```diff
@@ -259,28 +259,52 @@
...
 	new /obj/item/circuitboard/machine/plantgenes(src)// plant manipulator
 	new /obj/item/stock_parts/manipulator/femto(src)// for gibber, works with clothes on/faster.
 	new /obj/item/stock_parts/matter_bin/bluespace(src)// for gibber. gives most meat.
 
+/datum/gear/donator/kits/moribund_skills // characters teachings from my goat
+	name = "Moribund Teachings"
+	path = /obj/item/storage/box/large/custom_kit/moribund_skills
+	ckeywhitelist = list ("mottedesstriets")
+
+/obj/item/storage/box/large/custom_kit/moribund_skills/PopulateContents()
```
Comment (out-of-phaze) [inline modular_citadel/code/modules/client/loadout/__donator.dm:285]: No. Don't do this. Don't use `PopulateContents` for things other than populating contents. Make a custom granter book if you want, but don't do this.

File: modular_citadel/code/modules/client/loadout/__donator.dm [hunk 4]
```diff
@@ -259,28 +259,52 @@
...
+	var/player = get_mob_by_key("mottedesstriets") // I really like how clean this keeps things alffd so im borrowing it <3
+	if(!ishuman(player)) //This should never be true, but who knows.
+		return
+	var/mob/living/carbon/human/P = player
+	ADD_TRAIT(P, TRAIT_SURGERY_HIGH, src)// moribunds teachings to reduce the book clutter in the loadout box
+	ADD_TRAIT(P, TRAIT_UNETHICAL_PRACTITIONER, src)// ^
+	ADD_TRAIT(P, TRAIT_UNETHICAL_PRACTITIONER_BRAINWASHING, src)// ^
```
Comment (out-of-phaze) [inline modular_citadel/code/modules/client/loadout/__donator.dm:292]: This is given nowhere else in the game. I don't think it's balanced to give it to you roundstart, as any role.

File: code/modules/clothing/suits/medium_armor.dm [hunk 5]
```diff
@@ -402,6 +412,72 @@
...
+
+					M.adjustOxyLoss(-20, 0)
+					M.adjustToxLoss(-20, 0)
+					M.updatehealth()
+					var/tplus = world.time - M.timeofdeath
+					if(M.revive())
+						M.grab_ghost()
+						M.do_jitter_animation(10)
+						playsound(src,  'sound/machines/defib_zap.ogg', 50, 1, -1)
+						M.visible_message("<span class='green'>[M]'s suit makes a thump! followed by [M] taking his first breaths again!</span>")
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:469]:
```suggestion
						M.visible_message("<span class='green'>[M]'s suit makes a thump, followed by [M] taking [M.their()] first breaths again!</span>")
```

File: code/modules/clothing/suits/medium_armor.dm [hunk 6]
```diff
@@ -402,6 +412,72 @@
...
+/obj/item/clothing/suit/armored/medium/combat/flatline/prototype/process()
+	if(iscarbon(loc))
+		var/mob/living/carbon/M = loc
+		if(M.stat == DEAD || M.health < M.maxHealth)
+			M.adjustBruteLoss(-2) 
+			M.adjustFireLoss(-2)
+			M.adjustToxLoss(-2)
+			M.adjustOxyLoss(-2)
+			if(M.suiciding || M.hellbound) //they are never coming back
+				M.visible_message("<span class='warning'>[M]'s suit makes a long flatline noise...</span>")
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:442]:
```suggestion
				M.visible_message(span_warning("[M]'s suit makes a long flatline noise..."))
```

File: code/modules/clothing/suits/medium_armor.dm [hunk 7]
```diff
@@ -320,6 +320,16 @@
 // BROTHERHOOD OF STEEL //
 //////////////////////////
 
+/obj/item/clothing/suit/armored/medium/combat/flatline/prototype/bos
+	name = "Brotherhood 'Flatline' Armor"
+	desc = "A modified and refined self-revival 'flatline' armor developed by none other than flatline himself for his pack member he cares so much for. Repainted to the color scheme of the Brotherhood of Steel. Its lining inside is a gel-based refinement of the symbiotic algae found in buffalo gourds giving it flesh-mending properties over time. Inside it is a regulated and biomonitor-controlled self-defibrilator unit to bring the host back to life after the gel has done its job."
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:325]: Is "flatline" a name? If so, capitalize it.
```suggestion
	desc = "A modified and refined self-revival 'flatline' armor developed by none other than Flatline himself for his pack member he cares so much for. Repainted to the color scheme of the Brotherhood of Steel. Its lining inside is a gel-based refinement of the symbiotic algae found in buffalo gourds giving it flesh-mending properties over time. Inside it is a regulated and biomonitor-controlled self-defibrillator unit to bring the host back to life after the gel has done its job."
```

File: code/modules/clothing/suits/medium_armor.dm [hunk 8]
```diff
@@ -402,6 +412,72 @@
...
+			M.adjustBruteLoss(-2) 
+			M.adjustFireLoss(-2)
+			M.adjustToxLoss(-2)
+			M.adjustOxyLoss(-2)
+			if(M.suiciding || M.hellbound) //they are never coming back
+				M.visible_message("<span class='warning'>[M]'s suit makes a long flatline noise...</span>")
+				playsound(src, 'sound/machines/defib_failed.ogg', 50, 0)
+				return
+			if(M.getBruteLoss() >= 200 || M.getFireLoss() >= 200 || HAS_TRAIT(M, TRAIT_HUSK)) //body is too damaged to be revived
+				to_chat(M, "<span class='notice'>You feel your wounds sealing back toghether slowly!</span>")
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:446]:
```suggestion
				to_chat(M, span_notice("You feel your wounds sealing back together slowly!"))
```

File: code/modules/clothing/suits/medium_armor.dm [hunk 9]
```diff
@@ -402,6 +412,72 @@
...
 
+/obj/item/clothing/suit/armored/medium/combat/flatline/prototype
+	name = "prototype 'flatline' armor"
+	desc = "A very early model of the self-revival 'flatline' armor developed by none other than flatline himself. This one is highly customized. And by customized its cobbled together from junk, dripping algae and has a defib unit welded and installed on the chest piece. It looks like a scav made it."
+	icon_state = "flatline_prototype"
+	item_state = "flatline_prototype"
+	permeability_coefficient = 0.9
+	heat_protection = CHEST | GROIN | LEGS
+	cold_protection = CHEST | GROIN | LEGS
+	armor = list("melee" = 45, "bullet" = 35, "laser" = 35, "energy" = 35, "bomb" = 35, "bio" = 35, "rad" = 35, "fire" = 35, "acid" = 35, "wound" = 10)
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:423]: Why does the 'very early model' have more armor than the 'modified and refined' version?

File: code/modules/clothing/suits/medium_armor.dm [hunk 10]
```diff
@@ -402,6 +412,72 @@
...
+
+/obj/item/clothing/suit/armored/medium/combat/flatline/prototype/New()
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:425]: No. Don't use /New() for anything but datums.
```suggestion
/obj/item/clothing/suit/armored/medium/combat/flatline/prototype/Initialize()
```

File: code/modules/clothing/suits/medium_armor.dm [hunk 11]
```diff
@@ -402,6 +412,72 @@
...
+	..()
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:426]: Once you change this to initialize you'll need to return parent as well.
```suggestion
	. = ..()
```

File: code/modules/clothing/suits/medium_armor.dm [hunk 12]
```diff
@@ -402,6 +412,72 @@
 // CUSTOM //
 ////////////
 
+/obj/item/clothing/suit/armored/medium/combat/flatline/prototype
+	name = "prototype 'flatline' armor"
+	desc = "A very early model of the self-revival 'flatline' armor developed by none other than flatline himself. This one is highly customized. And by customized its cobbled together from junk, dripping algae and has a defib unit welded and installed on the chest piece. It looks like a scav made it."
```
Comment (out-of-phaze) [inline code/modules/clothing/suits/medium_armor.dm:417]:
```suggestion
	desc = "A very early model of the self-revival 'flatline' armor developed by none other than Flatline himself. This one is cobbled together from junk, dripping algae, and has a defib unit welded and installed on the chestpiece. It looks like a scav made it."
```

File: code/modules/clothing/suits/medium_armor.dm (hunk 7 above)
Comment (out-of-phaze) [reply code/modules/clothing/suits/medium_armor.dm:325]: Also, highly dislike having character backstory in item descriptions. How would someone know all this just from examining the armor? Especially if they've stolen it, or taken it off of a corpse?

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
Strongly opposed to a number of these changes for balance reasons. Some other minor concerns about code quality etc.

**Comment by out-of-phaze** [comment]
> ## Changelog
> 
> 🆑 added a 2 heal over time armor that works the same as the trauma backpack it auto defibs on death but takes alot of time for the slow heal to get them to revivable. and even longer before they can stand. revives on the boarder of dead. and takes a long time to heal out of soft crit armor stats are armor = list("melee" = 35, "bullet" = 35, "laser" = 30, "energy" = 20, "bomb" = 25, "bio" = 10, "rad" = 10, "fire" = 20, "acid" = 10, "wound" = 10) (same as initiate armor) and can only be created by scribes and higher rank scribes. tightened up my fluff box to be less book saturated because it was fucking with my OCD so I made it just auto-give the traits when selected /🆑

You also didn't even read the directions on how to use the changelog.
Also, the armor listed is wrong, or more accurately you don't mention *your* armor's stats at all.

**Comment by Moribun** [comment]
do you want to knock out the few people willing to code? I dont get payed for this I have no proper training. I do it because I was asked, have rudimentary knowledge and no one else was willing too. I have alot to say about you and im not gonna say any of them. but you are seriously making me not want to do this shit anymore.

**Comment by out-of-phaze** [comment]
> do you want to knock out the few people willing to code? I dont get payed for this I have no proper training. I do it because I was asked, have rudimentary knowledge and no one else was willing too. I have alot to say about you and im not gonna say any of them. but you are seriously making me not want to do this shit anymore.

I literally suggested changes; adding and committing them can be done from the "files changed" view. You just need to add them to the batch and then commit them. It's that easy.

The rest is balance concerns which I'm listing more as a matter of procedure for whoever decides if the PR is merged to look at.

**Comment by Moribun** [comment]
whatever im done

**Pull request closed without merging**